scripts/utils.py

```diff
@@ -73,10 +73,13 @@ def parse_sol_version(source):
                 return DEFAULT_SOLC_VERSION
             else:
                 solc_version = next(COMP_VERSION1_REX.finditer(l))
-                if solc_version not in SOLC_VERSIONS:
+                if solc_version is None:
```

Comment:
When can `solc_version` be None`?

Reply:
Yes sorry I thought `next()` returns `None` when the iterator is finished. It actually throws a `StopIteration` exception. So I guess there should be a check that this is not thrown when calling `next()`.

The reason is because the pragma might be after an empty line (or even just missing completely), in which case the returned iterator from `finditer()` will be empty, so calling `next()` will throw that exception.

Reply:
I guess using `COMP_VERSION_REX.search(l)` makes more sense anyway, since we only use the very first occurrence of a match. Then the `None` check would ensure a match was actually found.